File: mobile/src/api/brainDump.ts
import { apiRequest } from "./client";

export type BrainDumpSignals = {
  sentiment_score: number;
  emotions: string[];
  topics: string[];
  actionable_items: string[];
  acknowledgement: string;
};

export type BrainDumpResponse = {
  id: string;
  acknowledgement: string;
  signals: BrainDumpSignals;
  actionable: boolean;
};

export type BrainDumpRequest = {
  user_id: string;
  text: string;
};

export async function submitBrainDump(payload: BrainDumpRequest): Promise<{
  data: BrainDumpResponse;
  requestId: string | null;
}> {
  const { data, response } = await apiRequest<BrainDumpResponse>("/brain-dump", {
    method: "POST",
    body: payload,
  });

  return {
    data,
    requestId: response.headers.get("X-Request-Id"),
  };
}

File: backend/app/api/schemas/brain_dump.py
"""Pydantic schemas for brain dump API."""
from __future__ import annotations

from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class BrainDumpRequest(BaseModel):
    user_id: UUID
    text: str = Field(..., min_length=1, max_length=2000)


class BrainDumpSignals(BaseModel):
    sentiment_score: float = Field(..., ge=-1.0, le=1.0)
    emotions: List[str] = Field(default_factory=list)
    topics: List[str] = Field(default_factory=list)
    actionable_items: List[str] = Field(default_factory=list)
    acknowledgement: str


class BrainDumpResponse(BaseModel):
    id: UUID
    acknowledgement: str
    signals: BrainDumpSignals
    actionable: bool
